core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -357,4 +375,191 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     sc.conf.getTimeAsSeconds("spark.mesos.rejectOfferDurationForReachedMaxCores", "120s")
   }
 
+  /**
+   * Checks executor ports if they are within some range of the offered list of ports ranges,
+   *
+   * @param conf the Spark Config
+   * @param ports the list of ports to check
+   * @return true if ports are within range false otherwise
+   */
+  protected def checkPorts(conf: SparkConf, ports: List[(Long, Long)]): Boolean = {
+
+    def checkIfInRange(port: Long, ps: List[(Long, Long)]): Boolean = {
+      ps.exists(r => r._1 <= port & r._2 >= port)
+    }
+
+    val portsToCheck = nonZeroPortValuesFromConfig(conf)
+    val withinRange = portsToCheck.forall(p => checkIfInRange(p, ports))
+    // make sure we have enough ports to allocate per offer
+    ports.map(r => r._2 - r._1 + 1).sum >= portsToCheck.size && withinRange
+  }
+
+  /**
+   * Partitions port resources.
+   *
+   * @param portsToAssign non-zero ports to assign
+   * @param offeredResources the ports offered
+   * @return resources left, port resources to be used.
+   */
+  def partitionPortResources(portsToAssign: List[Long], offeredResources: List[Resource])
+    : (List[Resource], List[Resource]) = {
+    if (portsToAssign.isEmpty) {
+      (offeredResources, List[Resource]())
+    }
+    // partition port offers
```

Comment:
Then you need use `return` explicitly, or wrap the remaining code with an `else` block , otherwise it would continue to run the code after the `if` block.